solr/core/src/java/org/apache/solr/search/SolrConstantScoreQuery.java

```diff
@@ -41,17 +41,20 @@
  * Experimental and subject to change.
  */
 public class SolrConstantScoreQuery extends Query implements ExtendedQuery {
-  private final Filter filter;
```

Comment:
Out of scope of this issue, but SolrConstantScoreQuery can go away.  It's fundamental job is to make a Filter look like a Query way back when Filter wasn't a Query.  For many years, Filter has been one.  It also does what SolrExtendedQuery (or whatever it's called) does.

Reply:
Thanks for this info. I would be open to picking it up in another issue.

Reply:
Since this issue is now bigger in scope, it should remove SolrConstantScoreQuery.

Reply:
Justification:  SolrConstantScoreQuery's purpose *was* to bridge the Filter API to a Query API.  It became instant tech-debt the moment Filter subclassed Query (years ago).  Now with Filter going away, SCSQ has even less reason to exist -- it's basically a Lucene ConstantScoreQuery + Solr ExtendedQuery which we don't need.  Anyone who wants such a thing can simply do `new WrappedQuery(someOtherQuery)`

Reply:
Ok. I did some more work to remove SolrConstantScoreQuery. It's now up with my latest commit. Please give it a look, would love to get some feedback